ls1intum/Artemis
# Refactor connection status

### Checklist

### Motivation and Context
The connection status should also be used for working on team-based exercises. Right now, there is no component yet for this common use case; instead it exists multiple times as duplicates. This PR factors out this duplication and uses a shared component for all the occurences. This PR does not add anything new and should keep the exact same behavior as before.

### Description
- Create new component `JhiConnectionStatusComponent`
- Use new component in all existing use cases

### Steps for Testing
1. Log in to Artemis as Admin
2. Click on Server Administration
3. Click on Health
4. Check that the connection status in the row "Websocket Connection" is correctly displayed (it should very quickly switch from disconnected to connected on page load)
5. Set up a quiz exercise
6. Participate in the quiz exercise live
7. Check that the connection status is displayed correctly in the footer
8. Check as an instructor in the quiz statistics that the footer shows the connection status properly

### Screenshots
Just for reference (no visual change):
![OnPaste 20200331-191359](https://user-images.githubusercontent.com/7109225/78055417-c9b1bc80-7383-11ea-814c-a25d8db46d18.png)
